### ISSUE TITLE
WRN Missing key(s) in state_dict

### ISSUE THREAD

**Issue by gshdong** [ycszen/TorchSeg#45]
@ycszen 
When I run the train.py from cityscapes.bisenet.R18.speed, the following tip appears:
```
WRN Missing key(s) in state_dict: layer3.0.bn1.num_batches_tracked, layer1.1.bn1.num_batches_tracked, layer2.1.bn2.num_batches_tracked, layer1.1.bn2.num_batches_tracked, layer1.0.bn1.num_batches_tracked, layer2.0.downsample.1.num_batches_tracked, layer3.1.bn2.num_batches_tracked, layer3.1.bn1.num_batches_tracked, layer3.0.downsample.1.num_batches_tracked, layer2.0.bn1.num_batches_tracked, layer2.0.bn2.num_batches_tracked, layer4.0.bn1.num_batches_tracked, layer4.0.bn2.num_batches_tracked, bn1.num_batches_tracked, layer4.1.bn2.num_batches_tracked, layer4.1.bn1.num_batches_tracked, layer1.0.bn2.num_batches_tracked, layer3.0.bn2.num_batches_tracked, layer4.0.downsample.1.num_batches_tracked, layer2.1.bn1.num_batches_tracked
```
How should I deal with this problem?

**Comment by yaoqi-zd** [comment]
meet the same problem
```
02 19:49:05 WRN Unexpected key(s) in state_dict: heads.0.conv_3x3.conv.weight, heads.0.conv_3x3.bn.bias, heads.1.conv_3x3.bn.num_batches_tracked, heads.0.conv_3x3.bn.num_batches_tracked, heads.0.conv_1x1.bias, heads.0.conv_3x3.bn.running_var, heads.1.conv_1x1.bias, heads.1.conv_3x3.bn.weight, heads.1.conv_3x3.conv.weight, heads.1.conv_1x1.weight, heads.0.conv_3x3.bn.running_mean, heads.1.conv_3x3.bn.running_mean, heads.0.conv_3x3.bn.weight, heads.1.conv_3x3.bn.running_var, heads.1.conv_3x3.bn.bias, heads.0.conv_1x1.weight
```

have you solved it ?

**Comment by xyiyy** [comment]
@yaoqi-zd 
```
        if is_training:
            heads = [BiSeNetHead(conv_channel, out_planes, 2, # 16
                                 True, norm_layer),
                     BiSeNetHead(conv_channel, out_planes, 1,  # 8 
                                 True, norm_layer),
                     BiSeNetHead(conv_channel * 2, out_planes, 1, # 8
                                 False, norm_layer)]
        else:
            heads = [None, None,
                     BiSeNetHead(conv_channel * 2, out_planes, 1, # 8
                                 False, norm_layer)]
```
The author removes two auxiliary heads in evaluation.

**Comment by yaoqi-zd** [comment]
@xyiyy Thanks a lot!

**Comment by Jingchensun** [comment]
> @ycszen
> When I run the train.py from cityscapes.bisenet.R18.speed, the following tip appears:
> 
> ```
> WRN Missing key(s) in state_dict: layer3.0.bn1.num_batches_tracked, layer1.1.bn1.num_batches_tracked, layer2.1.bn2.num_batches_tracked, layer1.1.bn2.num_batches_tracked, layer1.0.bn1.num_batches_tracked, layer2.0.downsample.1.num_batches_tracked, layer3.1.bn2.num_batches_tracked, layer3.1.bn1.num_batches_tracked, layer3.0.downsample.1.num_batches_tracked, layer2.0.bn1.num_batches_tracked, layer2.0.bn2.num_batches_tracked, layer4.0.bn1.num_batches_tracked, layer4.0.bn2.num_batches_tracked, bn1.num_batches_tracked, layer4.1.bn2.num_batches_tracked, layer4.1.bn1.num_batches_tracked, layer1.0.bn2.num_batches_tracked, layer3.0.bn2.num_batches_tracked, layer4.0.downsample.1.num_batches_tracked, layer2.1.bn1.num_batches_tracked
> ```
> 
> How should I deal with this problem?

i get the same problem as you

**Comment by ycszen** [comment]
This is a log for loading checkpoint.
The new version of PyTorch adds `num_batches_tracked` parameters in the batch normalization function, which is not contained before. 
Therefore, it outputs this log. However, this didn't affect the training performance.
